### PR TITLE
MAYA-126625 Connections now handled internally

### PR DESCRIPTION
**Pull request by JGamache-autodesk**

### CODE REVIEW

File: lib/mayaUsd/ufe/UsdUndoConnectionCommands.cpp [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+    auto item
+        = std::dynamic_pointer_cast<UsdSceneItem>(Ufe::Hierarchy::createItem(attrInfo.path()));
+    if (!item) {
+        TF_RUNTIME_ERROR("Invalid scene item.");
+        return nullptr;
+    }
+    return UsdAttributes(item).attribute(attrInfo.name());
+}
+
+UsdAttribute* usdAttrFromUfeAttr(const Ufe::Attribute::Ptr& attr)
```
Comment (JGamache-autodesk) [inline lib/mayaUsd/ufe/UsdUndoConnectionCommands.cpp:51]: Code copied from UsdConnectionHandler.cpp with small adaptation to return the connection result.

File: test/lib/ufe/testConnections.py [hunk 2]
```diff
@@ -452,6 +453,109 @@ def testConnectionsHandler(self):
...
+        self.assertIsNotNone(connections)
+        conns = connections.allConnections()
+        self.assertEqual(len(conns), 1)
+
+        # Delete the connection with the new API command.
+
+        src = srcAttr.attribute()
+        dst = dstAttr.attribute()
+
+        cmd = connectionHandler.deleteConnectionCmd(src, dst)
```
Comment (JGamache-autodesk) [inline test/lib/ufe/testConnections.py:513]: Identical to previous implementation directly above, but using new API.

File: test/lib/ufe/testConnections.py [hunk 3]
```diff
@@ -571,6 +675,135 @@ def testCreateStandardSurface(self):
...
+
+        connectionCommand = connectionHandler.createConnectionCmd(shaderOutput, materialOutput)
+        connectionCommand.execute()
+
+        def assertNewConnectionMatchesCommand(self, materialItem):
+            connections = connectionHandler.sourceConnections(materialItem)
+            self.assertIsNotNone(connections)
+            conns = connections.allConnections()
+            self.assertEqual(len(conns), 1)
+            # Check the command result:
```
Comment (JGamache-autodesk) [inline test/lib/ufe/testConnections.py:757]: We can now find out about connections being remapped by looking at the command result.

File: test/lib/ufe/testConnections.py [hunk 4]
```diff
@@ -571,6 +675,135 @@ def testCreateStandardSurface(self):
...
+            self.assertEqual(conns[0].src.path, connectionCommand.connection.src.path)
+            self.assertEqual(conns[0].src.name, connectionCommand.connection.src.name)
+            self.assertEqual(conns[0].dst.path, connectionCommand.connection.dst.path)
+            self.assertEqual(conns[0].dst.name, connectionCommand.connection.dst.name)
+            self.assertEqual(connectionCommand.connection.dst.name, "outputs:mtlx:surface")
+        assertNewConnectionMatchesCommand(self, materialItem)
+
+        connectionCommand.undo()
```
Comment (JGamache-autodesk) [inline test/lib/ufe/testConnections.py:765]: Undo/Redo fully works, even with remappings.

File: lib/mayaUsd/ufe/UsdUndoConnectionCommands.h [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+//
+#ifndef MAYAUSD_UFE_USDUNDOCONNECTIONCOMMANDS_H
+#define MAYAUSD_UFE_USDUNDOCONNECTIONCOMMANDS_H
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/undo/UsdUndoableItem.h>
+
+#include <ufe/attribute.h>
+#include <ufe/connection.h>
+#include <ufe/path.h>
```
Comment (ppt-adsk) [inline lib/mayaUsd/ufe/UsdUndoConnectionCommands.h:24]: Don't think we need this one.

File: lib/mayaUsd/ufe/UsdUndoConnectionCommands.h [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+// limitations under the License.
+//
+#ifndef MAYAUSD_UFE_USDUNDOCONNECTIONCOMMANDS_H
+#define MAYAUSD_UFE_USDUNDOCONNECTIONCOMMANDS_H
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/undo/UsdUndoableItem.h>
+
+#include <ufe/attribute.h>
+#include <ufe/connection.h>
```
Comment (ppt-adsk) [inline lib/mayaUsd/ufe/UsdUndoConnectionCommands.h:23]: Do we need this one?

**Review by ppt-adsk — APPROVED** [review]
LGTM, except for minor include file tweaks.

File: lib/mayaUsd/ufe/UsdUndoConnectionCommands.h (hunk 5 above)
Comment (JGamache-autodesk) [reply lib/mayaUsd/ufe/UsdUndoConnectionCommands.h:24]: Not necessary. I removed it and cleaned up two comments.

**Review by ppt-adsk — APPROVED** [review]

**Review by hodoulp — DISMISSED** [review]
Only few comments which could be fixed in a followup pull request.

**Review by hodoulp — APPROVED** [review]

**Review by hodoulp — APPROVED** [review]

**Comment by JGamache-autodesk** [comment]
Marking as ready for merge. The 2 failures are from build machines unable to talk to Artifactory.

**Pull request merged**